### ISSUE TITLE
[Feature] add an option to override the DefaultExtractor 

### ISSUE THREAD

**Issue by ctf0** [spatie/laravel-mix-purgecss#40]
https://github.com/spatie/laravel-mix-purgecss/blob/c48b79335d1654f9f0d397c88ae602cf53fa14dc/src/createPurgeCssPlugin.js#L7

**Comment by cjmaxik** [comment]
Uuummm what is the justification?

**Comment by ctf0** [comment]
incase the user wants a different pattern to match against ex. dont match classes with `: or /` in its names

**Comment by cjmaxik** [comment]
I don't this anyone will be using that thing anyway. Also, there are `whitelistPatterns` to do so: https://www.purgecss.com/configuration#options

**Comment by ctf0** [comment]
why solve an issue but using a different option ? 

anyway here is the defaults https://www.purgecss.com/extractors#default-extractor which might be what someone might exactly need

i can make a PR for it if thats the issue.

**Comment by cjmaxik** [comment]
Hmmm... Can we use the DefaultExtractor of underlying library rather w/o using our own?

purgecss default: 
`return content.match(/[A-Za-z0-9_-]+/g) || []`

our default for now:
`return content.match(/[a-zA-Z0-9-:_/]+/g) || [];`

**Comment by ctf0** [comment]
wont change much, however can u tell me why you dont want to expose the option to the user ?

**Comment by aryehraber** [comment]
@cjmaxik The extractor has been tweaked slightly to support Tailwind out of the box, it even says so in the README:

> The Purgecss config is inspired by Jonathan Reinink's config for Tailwind CSS.

As for the FR itself to override the default, sounds fair enough though I'm sure most won't need it.

**Comment by sebastiandedeyne** [comment]
You can already override _any_ Purgecss setting. In this case, you need to override the `extractors` key entirely. 

I don't think I want to add an extra `extractor` option, would rather keep the amount of options specific to this wrapper as small as possible.

**Comment by adamwathan** [comment]
I would personally love if we could just provide our own regex. I often create extra classes with Tailwind that include characters like `%` or `.` for example and right now those don't get caught. Would be awesome if I could just provide a regex 🤙

**Comment by aryehraber** [comment]
@adamwathan Just submitted a [PR](https://github.com/spatie/laravel-mix-purgecss/pull/61) for this feature!

It'd be great if you could give it a test run for a quick sanity check. You can download the [updated plugin here](https://github.com/aryehraber/laravel-mix-purgecss) and plop it into your project root for now.

Usage: 

```js
const mix = require('laravel-mix')
const tailwindcss = require('tailwindcss')

require('./laravel-mix-purgecss') // note: relative path to new version

mix.postCss('resources/css/main.css', 'public/css', [
    tailwindcss('tailwind.js'),
  ])
  .purgeCss({
    extractorPattern: /[a-zA-Z0-9-:%_/]+/g
  })
```

**Comment by sebastiandedeyne** [comment]
Released in `4.1.0`!
